File: FluentFTP.VBExamples/DirectoryExists.vb
Imports System
Imports System.Net
Imports System.Threading
Imports System.Threading.Tasks
Imports FluentFTP

Namespace Examples
	Friend Module DirectoryExistsExample
		Sub DirectoryExists()
			Using conn = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				conn.Connect()

				If conn.DirectoryExists("/full/or/relative/path") Then
					' do something
				End If

			End Using
		End Sub

		Async Function DirectoryExistsAsync() As Task
			Dim token = New CancellationToken()

			Using conn = New AsyncFtpClient("127.0.0.1", "ftptest", "ftptest")
				Await conn.Connect(token)

				If Await conn.DirectoryExists("/full/or/relative/path") Then
					' do something
				End If

			End Using
		End Function
	End Module
End Namespace

File: FluentFTP.CSharpExamples/DirectoryExists.cs
using System;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using FluentFTP;

namespace Examples {
	internal static class DirectoryExistsExample {

		public static void DirectoryExists() {
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				conn.Connect();

				if (conn.DirectoryExists("/full/or/relative/path")) {
					// do something
				}
			}
		}

		public static async Task DirectoryExistsAsync() {
			var token = new CancellationToken();
			using (var conn = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				await conn.ConnectAsync(token);

				if (await conn.DirectoryExistsAsync("/full/or/relative/path", token)) {
					// do something
				}
			}
		}
	}
}